commit edd4cbcf83fb466dd8a1b5d63d55b1fea96356fe
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Fri Aug 7 01:38:39 2015 +0200

    Fix gcrt0.o compilation
    
    When static-start-installed-name is different from
    start-installed-name, we must not use the shared objects.
    
    * csu/Makefile
    (extra-objs): Add gmon-start.o when building shared library and
    $(static-start-installed-name) is different from
    $(start-installed-name).
    $(objpfx)g$(static-start-installed-name): When building shared
    library and $(static-start-installed-name) is different from
    $(static-start-installed-name), revert to non-shared rule,
    i.e. using $(objpfx)% and gmon-start.o.

diff --git a/csu/Makefile b/csu/Makefile
index 9f0855a..4df5d10 100644
--- a/csu/Makefile
+++ b/csu/Makefile
@@ -47,6 +47,9 @@ tests-static := tst-empty
 
 ifeq (yes,$(build-shared))
 extra-objs += S$(start-installed-name) gmon-start.os
+ifneq ($(start-installed-name),$(static-start-installed-name))
+extra-objs += gmon-start.o
+endif
 install-lib += S$(start-installed-name)
 generated += start.os
 else
@@ -98,10 +101,14 @@ endif
 # startfile with gmon-start.o, which defines a constructor function
 # to turn on profiling code at startup.
 ifeq (yes,$(build-shared))
-$(addprefix $(objpfx),$(sort g$(start-installed-name) \
-			     g$(static-start-installed-name))): \
+$(objpfx)g$(start-installed-name): \
   $(objpfx)g%: $(objpfx)S% $(objpfx)gmon-start.os
 	$(link-relocatable)
+ifneq ($(start-installed-name),$(static-start-installed-name))
+$(objpfx)g$(static-start-installed-name): \
+  $(objpfx)g%: $(objpfx)% $(objpfx)gmon-start.o
+	$(link-relocatable)
+endif
 else
 $(addprefix $(objpfx),$(sort g$(start-installed-name) \
 			     g$(static-start-installed-name))): \
